warehouse/templates/includes/file-details.html

```diff
@@ -147,16 +155,26 @@ <h3>Provenance</h3>
                 <li>Owner: <a href="{{ viewer.owner }}">{{ viewer.owner }}</a></li>
                 <li>Access: <code>{{ viewer.access }}</code></li>
               </ul>
+              {% endif %}
               Publication detail:
               <ul>
                 <li>Token Issuer: <code>{{ viewer.issuer }}</code></li>
+                {% if viewer.publisher.kind == 'Google' %}
+                <li>Service Account: <code>{{ viewer.publisher.email }}</code></li>
+                {% endif %}
+                {% if viewer.environment %}
                 <li>Runner Environment: <code>{{ viewer.environment }}</code></li>
+                {% endif %}
+                {% if viewer.workflow_filename and viewer.build_digest %}
                 <li>Publication workflow:
                   <a href="{{ viewer.workflow_url }}">
                     <code>{{ viewer.workflow_filename }}@{{ viewer.build_digest }}</code>
                   </a>
                 </li>
+                {% endif %}
+                {% if viewer.tigger %}
```

Comment:
![](https://media0.giphy.com/media/v1.Y2lkPTc5MGI3NjExdTltdXExdGNlOGllNWlwdDFkYjVkODR5NnhiM21oa3h1OGF0MXZ5YyZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/RESoRHgqFeSAg/giphy.gif)
```suggestion
                {% if viewer.trigger %}
```